lib/libutee/tee_api_objects.c

```diff
@@ -38,24 +38,87 @@
 #define TEE_ATTR_BIT_PROTECTED              (1 << 28)
 
 /* Data and Key Storage API  - Generic Object Functions */
+/*
+ * Use of this function is deprecated
+ * new code SHOULD use the TEE_GetObjectInfo1 function instead
+ * These functions will be removed at some future major revision of
+ * this specification
+ */
 void TEE_GetObjectInfo(TEE_ObjectHandle object, TEE_ObjectInfo *objectInfo)
 {
 	TEE_Result res;
 
-	res = utee_cryp_obj_get_info((uint32_t)object, objectInfo);
+	res = TEE_GetObjectInfo1(object, objectInfo);
```

Comment:
`TEE_GetObjectInfo()` will always panic on corrupted objects when implemented like this.